plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiUtil.java

```diff
@@ -271,4 +287,90 @@ public static HoodieTableFileSystemView getFileSystemView(
         return new HoodieTableFileSystemView(
                 tableMetadata, metaClient, metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants());
     }
+
+    public static HiveHudiPartitionInfo buildHiveHudiPartitionInfo(HudiTableHandle tableHandle, String partitionName, Partition partition)
+    {
+        return new HiveHudiPartitionInfo(
+                tableHandle.getSchemaTableName(),
+                Location.of(tableHandle.getBasePath()),
+                partitionName,
+                partition,
+                tableHandle.getPartitionColumns(),
+                tableHandle.getPartitionPredicates());
+    }
+
+    public static Partition buildPartition(String partitionPath, List<Column> partitionColumns, HudiTableHandle tableHandle, PartitionValueExtractor partitionValueExtractor)
```

Comment:
Same on moving this to `HudiBackgroundSplitLoader` and make it private/default visibility.